homeassistant/components/recorder/statistics.py

```diff
@@ -1968,8 +1992,8 @@ def no_conversion(val: float | None) -> float | None:
         stats,
```

Comment:
We do this groupby twice so we end up iterating all the states twice.  I think its going to be faster to save the list from the first groupby in a dict and use it below to avoid iterating everything twice.  Everything ends up in a dict anyways at the end so it shouldn't be a memory issue.